### PR TITLE
Add for_key() and for_value()

### PR DESCRIPTION
**Pull request by dmolesUC3**
See issue #6

### CODE REVIEW

**Comment by dblock** [comment]
I like the implementation, but I don't love the naming. What if we just called these `key` and `value`?

This also needs an entry in CHANGELOG and README documentation, please.

**Comment by dblock** [comment]
Also I think `to_h` can now be rewritten in terms of one of the memoized variables, but needs to be `dup`-ed.

**Comment by dblock** [comment]
Also see https://github.com/alterego-labs/ruby-enum/commit/5c4aff2e8439d882e1131187146d51dd642b6763 which implements what should be `key?` and `value?`, both I think are also useful.

**Comment by dmolesUC3** [comment]
Makes sense. I'll do the renames and (re)implement key? and value?. Do you want me to update to_h in the same pull request?

**Comment by dblock** [comment]
You should update to_h because it will use your new code, too.

Please don't forget to squash your commits.

**Comment by dmolesUC3** [comment]
Hmm. As I'm working on additions to the README, I realize I've implemented `key?` to take a symbol but implemented `key` and `value` to return enum instances.

```ruby
Colors.key?(:RED)
# => true 
Colors.key('red')
# => #<Colors:0x007fbdb49ed620 @key=:RED, @value="red">
Colors.value?('green')
# => true
Colors.value(:GREEN)
# => #<Colors:0x007fbdb49ed4e0 @key=:GREEN, @value="green"> 
```

Since `keys` returns symbols and `values` returns strings, and since AFAICT there's not actually any way to get to the enum instance except with `each`, this probably isn't the desired behavior. I'll modify `key` and `value` to be consistent with `keys` and `values` and `to_h`.

And I'll update `to_h` and squash the commits.

**Comment by dmolesUC3** [comment]
Okay, done. I didn't end up updating `to_h` -- maybe I didn't understand what you were suggesting, but `@_enum_hash` and `@_enums_by_value` still both use enum instances, and `to_h` still needs to extract values, so the existing implementation looks correct to me.

File: CHANGELOG.md [hunk 1]
```diff
@@ -1,6 +1,6 @@
 ### Next
 
-* Your contribution here.
```
Comment (dblock) [inline CHANGELOG.md]: Put this back :)

File: CHANGELOG.md [hunk 2]
```diff
@@ -1,6 +1,6 @@
...
+* [#8](https://github.com/dblock/ruby-enum/pull/8): Added `Ruby::Enum#key`, `Ruby::Enum#value`, `Ruby::Enum#key?`, and `Ruby::Enum#value?` - [@dmolesUC3](https://github.com/dmolesUC3)
```
Comment (dblock) [inline CHANGELOG.md]: Add a period to the end of this sentence. 

Nitpicky me ;)

File: spec/ruby-enum/enum_spec.rb [hunk 3]
```diff
@@ -7,6 +7,17 @@ class Colors
   define :GREEN, 'green'
 end
 
+# A test class that includes CONSTANT_CASE with underscores
```
Comment (dblock) [inline spec/ruby-enum/enum_spec.rb]: Can I nitpick on this test? Why do we need `X11Colors` at all and can't write it all with the existing `Colors` structure? Less code, more readable, no?

**Comment by dblock** [comment]
The implementation is :+1:, see my other comments? Thanks!

**Comment by dmolesUC3** [comment]
Sorry, `X11Colors` is a relic from when I thought there was a bug relating to underscores in keys. I'll take it out and add the contribution line back.

**Comment by dmolesUC3** [comment]
Done.

**Comment by dblock** [comment]
Perfect, merging.

**Pull request merged**